plugins/module_utils/errors.py

```diff
@@ -0,0 +1,86 @@
+# -*- coding: utf-8 -*-
+
+# Copyright: Contributors to the Ansible project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+import functools
+
+try:
+    import botocore
+except ImportError:
+    pass  # Modules are responsible for handling this.
+
+from .exceptions import AnsibleAWSError
+
+
+class AWSErrorHandler:
+    """_CUSTOM_EXCEPTION can be overridden by subclasses to customize the exception raised"""
+
+    _CUSTOM_EXCEPTION = AnsibleAWSError
+
+    @classmethod
+    def _is_missing(cls):
+        """Should be overridden with a class method that returns the value from is_boto3_error_code (or similar)"""
+        return type("NeverEverRaisedException", (Exception,), {})
+
+    @classmethod
+    def common_error_handler(cls, description):
+        """A simple error handler that catches the standard Boto3 exceptions and raises
+        an AnsibleAWSError exception.
+        """
+
+        def wrapper(func):
+            @functools.wraps(func)
+            def handler(*args, **kwargs):
+                try:
+                    return func(*args, **kwargs)
+                except botocore.exceptions.WaiterError as e:
+                    raise cls._CUSTOM_EXCEPTION(message=f"Timeout trying to {description}", exception=e) from e
+                except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+                    raise cls._CUSTOM_EXCEPTION(message=f"Failed to {description}", exception=e) from e
+
+            return handler
+
+        return wrapper
+
+    @classmethod
+    def list_error_handler(cls, description, default_value=None):
+        """A simple error handler that catches the standard Boto3 exceptions and raises
+        an AnsibleAWSError exception.
+        Error codes representing a non-existent entity will result in None being returned
```

Comment:
Could we update this docstring to note that `default_value` can be passed to change the return value to, for example, an empty list? This is a clever feature that I think would be easy for developers to miss.

Reply:
done